nexus-repository-composer/src/main/java/org/sonatype/nexus/repository/composer/internal/ComposerJsonProcessor.java

```diff
@@ -351,7 +351,7 @@ public Content mergePackagesJson(final Repository repository, final List<Payload
     for (Payload payload : payloads) {
       Map<String, Object> json = parseJson(payload);
       Map<String, Object> providers = (Map<String, Object>) json.get(PROVIDERS_KEY);
-      names.addAll(providers.keySet());
+      Optional.ofNullable(providers).ifPresent(pv -> names.addAll(pv.keySet()));
```

Comment:
Why not using a simple old-school but cheap null check instead of creating an Optional instance and an non-static instance lambda?
```suggestion
      if (providers != null) {
          names.addAll(providers.keySet());
      }
```

If you insist on functional style, even this would be more efficient:
```java
Optional.ofNullable(providers).map(Map::keySet).ifPresent(names::addAll);
```